pkg/controller/virtualmachinevolumeexport/pvc.go

```diff
@@ -0,0 +1,109 @@
+package virtualmachinevolumeexport
+
+import (
+	"context"
+	goerrors "errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	hc "kubevirt-image-service/pkg/apis/hypercloud/v1alpha1"
+	vmv "kubevirt-image-service/pkg/controller/virtualmachinevolume"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *ReconcileVirtualMachineVolumeExport) syncPvc() error {
+	if _, err := r.getPvc(GetPvcName(r.vmvExport.Name)); err == nil {
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return err
+	}
+
+	// get virtual machine volume pvc
+	sourcePvc, err := r.getPvc(vmv.GetVolumePvcName(r.vmvExport.Spec.VirtualMachineVolume.Name))
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Create a new pvc for vmvExport %s", r.vmvExport.Name)
+	if err2 := r.updateStateWithReadyToUse(r.vmvExport, false, hc.VirtualMachineVolumeExportStateCreating); err2 != nil {
```

Comment:
why err2?

Reply:
It has lint error.
`shadow: declaration of "err" shadows declaration at line 26`

Reply:
That's weird. how about this?
```
if err = r.(); err != nil {

}
```

Reply:
I've tried it too, but I still get the lint error.

sloppyReassign: re-assignment to `err` can be replaced with `err := r.updateStateWithReadyToUse(r.vmvExport, false, hc.VirtualMachineVolumeExportStateCreating)` (gocritic)